### PR TITLE
Refactoring & 	Revert to the old way of displaying the list of tables

### PR DESCRIPTION
**Pull request by ruslantalpa**
This is needed to allow the possibility of 1 user = 1 db role scenario.

### CODE REVIEW

**Comment by begriffs** [comment]
What caused the table listing change earlier, and this reversion? Is it possible to make a test to illustrate the change?

Also I know we pulled the PUT handler out earlier. I wasn't sure if it was coming back in a refactored form, but the removal from cors makes me think you'd rather eliminate it. Is your reasoning that it's a rare case and doesn't work well with bulk actions?

Thanks for updating the debian stuff too.

**Comment by calebmer** [comment]
What is the big picture change here? Locally, I'm working on some stuff to make view relation detection a lot better, and this would break some things for me.

**Comment by ruslantalpa** [comment]
@begriffs the table listing change was triggered by the fact that in scenarios where you have 1user = 1role the allTables query breaks down because for every table, it needs a list or roles that can access it.
For the put stuff, i thought leave the pending tests as a reminder but since there is no put functionality at the moment, remove it from cors. The reasoning for eliminating is cost/benefit, i have not seed a good explanation what you gain from upsert in postgrest and the cost is that the "upsert" query is very different in structure then the other ones and it complicates the code a lot + the way it is done is not 100% clean.

@calebmer what problems does this cause for you (eliminating the list of tables). What view relations are not detected?

**Comment by calebmer** [comment]
@ruslantalpa views in different schemas with different table names and with different column names are not detected. I'll explain the problem domain better in my PR 😊

In that case could we just eliminate the role detection for the table instead of eliminating the tables from `DbStructure` entirely?

**Comment by ruslantalpa** [comment]
if you eliminate that then users will see more tables than they have access to.

**Comment by calebmer** [comment]
Sorry, I meant only move the role detection instead of moving table detection, not eliminate it entirely.

**Comment by begriffs** [comment]
@calebmer should we hold off merging this then?

I do think we have some refactoring to do in App.hs at some point. It's getting hard to understand. Some routes use the request parser, some don't. The `parseRequest` function itself is kind of bewildering. It does a lot and returns a triple. Some of the routes don't really use all the parts of the triple which makes me think those parts could be separated into their own functions. Don't get me wrong I'm super grateful for the new functionality provided by this code, but I think it can get cleaned up.

**Comment by ruslantalpa** [comment]
This PR (the first commit) is one step towards cleaning up App.hs, i agree it's gotten a bit unclear/complicated (maybe we all can have a chat, i'll explain the logic in it then everybody can think of ways to simplify it).
About holding this PR, i think it should be merged and the "breaking changes" that affect @calebmer code i think he can easily address on his side. What this pr removed is a list of all the tables in the database (because it was not used anywhere else then / path). @calebmer seems to need this list to detect some other relations. Since this list will exclusively be used to generate  the "relations" list, it can be called/created somewhere inside https://github.com/begriffs/postgrest/blob/v3/src/PostgREST/PgStructure.hs#L110 And also, the query that generates the tables list can be simplified a lot on the account that the code does not need the list of roles that have access to the tables.

In conclusion, @begriffs please merge the PR :) @calebmer get the list of tables inside "allRelations" function

**Comment by ruslantalpa** [comment]
@calebmer sorry i made your task a bit more complicated :) but it's for the best. I can help you out withe the query for the table list if you need
